src/rollup/types.d.ts

```diff
@@ -425,6 +441,9 @@ export interface FunctionPluginHooks {
 			format: InternalModuleFormat;
 			moduleId: string;
 			targetModuleId: string | null;
+			chunk: PreRenderedChunkWithFilename;
```

Comment:
I was just checking the code and I wonder if you couldn't even use RenderedChunk here. This one already contains the fileName, though it is technically preliminary as you can see from the documentation here https://rollupjs.org/plugin-development/#renderchunk. Then you do not need to introduce a new type.
This one also includes `chunk.imports`, which is a list of dependencies as preliminary file names. Of course, it does not explicitly distinguish between internal and external dependencies.

Reply:
Calling `Chunk.getRenderedChunkInfo()` from `ImportExpression.setExternalResolution` seems to fail 199 tests. I guess some values aren't ready yet and since it's cached, it ends up causing some problems later on. I assume reusing that type could be a bit confusing if all the values aren't available.

I just realized that the casing for `preliminaryFilename` is inconsistent with another field `OutputChunk.preliminaryFileName` (from `generateBundle`). I'll fix that, unless perhaps you would like this to just be `fileName` instead?

Reply:
> seems to fail 199 tests

I see, then we leave it as it is.

> unless perhaps you would like this to just be fileName instead?

Actually, maybe we should switch. While `preliminaryFilename` is more correct, in all other places we just call preliminary file names `fileName`. Also, this would allow to later switch to RenderedChunk if we figure out what the problem was.